### DISCUSSION TITLE
Multiple instances on same server?

### DISCUSSION THREAD

**Discussion by dylan-meiners** [coder/code-server#2280] [Q&A] [answered]
I have been looking for a while on ways to have multiple code-servers running at the same time. It would really be a hassle to have to have a different VM for each code-server instance. As far as I can tell, running code-server always uses port 8080. I read somewhere that if you wanted to have different team members, you would need multiple instances. How would I start another instance of code-server on a different port?

**Comment by code-asher** [comment, marked as answer]
You can use the --bind-addr or --port flags. For example:

code-server --bind-addr localhost:1234
code-server --port 1234

You can also use port 0 to use a random port.

One thing to keep in mind is that if code-server is handling TLS then
there is a bug with running multiple instances: https://github.com/cdr/code-server/issues/2200

Using a reverse proxy to handle TLS won't have any problems though.

**Comment by bpmct** [comment]
@dylan-meiners curious how you ended up doing this. I have been using containers and am thinking of setting up a minikube cluster to give some friends their own web dev environments :)

**Reply by dylan-meiners** [reply]
My current plan is to setup a server with multiple Docker instances running the official code-server image for them. I will let you know how it goes. This would work because I could direct certain ports from outside to the correct Docker image's port 8080.

**Reply by dylan-meiners** [reply]
I found [this](https://dzone.com/articles/minikube-cloud-code-vscode) that might help you too.

**Comment by dylan-meiners** [comment]
I will leave this thread open, in case someone else has any solutions that they used, as there are many to choose from.

So far, the list is

- Running multiple servers directly, but specifying a different port for each _@code-asher_ 
- Running Docker containers with the code-server image _@rafket_ _[here](https://github.com/cdr/code-server/issues/356#issuecomment-478740818)_
- Running Minikubes with a code-server instance on each _@bpmct_
